### ISSUE TITLE
Changing `simpl never` not respected with alias

### ISSUE THREAD

**Issue by SkySkimmer** [coq/coq#18490]
Remnant of #13428 
~~~coq
Definition plus' := plus.

Definition foo := Eval simpl in plus' 1 2.

Arguments plus : simpl never.

Lemma test P : P 3 -> P (plus' 1 2).
Proof.
intros p3.
simpl.
(* goal is P 3 *)
~~~
This is because `Definition foo` precomputed the unfolding of `plus'` with `plus` not simpl never and cached it in a global table https://github.com/coq/coq/blob/068232d2c884b6ac829d800e417682ba57b3ca6a/pretyping/tacred.ml#L177

In other words the caching at https://github.com/coq/coq/blob/068232d2c884b6ac829d800e417682ba57b3ca6a/pretyping/tacred.ml#L352-L362 is not correct when `allowed_reds` changes.
